evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    any_values_are_nan,
+    keep_non_nan_rows,
+    pad_with_nans
+)
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    """Pipeline base class for time series regression problems."""
+
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series regression problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified as a dictionary with the "pipeline" key.
+            random_state (int, np.random.RandomState): The random seed/state. Defaults to 0.
+        """
+        if "pipeline" not in parameters:
+            raise ValueError("gap and max_delay parameters cannot be omitted from the parameters dict. "
+                             "Please specify them as a dictionary with the key 'pipeline'.")
+        pipeline_params = parameters.pop("pipeline")
+        self.gap = pipeline_params['gap']
+        self.max_delay = pipeline_params['max_delay']
+        super().__init__(parameters, random_state)
+
+    def fit(self, X, y):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+        X = _convert_woodwork_types_wrapper(X.to_pandas())
+        y = _convert_woodwork_types_wrapper(y.to_pandas())
```

Comment:
Nice! @angela97lin does this look good to you?

Reply:
Yup, looks good!